From 5f5a5dca532aa76de5edf323d6e455df4abc9345 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:33 -0500
Subject: [ethernet] r8169: fix setting rx vlan

Message-id: <1416413687-12219-3-git-send-email-ivecera@redhat.com>
Patchwork-id: 100350
O-Subject: [RHEL7.1 PATCH 02/16] r8169: fix setting rx vlan
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit 36d8e82541c4be8f529dce9efbcefd504e085ccf
Author: hayeswang <hayeswang@realtek.com>
Date:   Fri Sep 12 11:35:12 2014 +0800

    r8169: fix setting rx vlan

    The setting should depend on the new features not the current one.

    Signed-off-by: Hayes Wang <hayeswang@realtek.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index d54e6c8..ace4cd2 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -1860,7 +1860,7 @@ static void __rtl8169_set_features(struct net_device *dev,
   else
    tp->cp_cmd &= ~RxChkSum;
 
-  if (dev->features & NETIF_F_HW_VLAN_CTAG_RX)
+  if (features & NETIF_F_HW_VLAN_CTAG_RX)
    tp->cp_cmd |= RxVlan;
   else
    tp->cp_cmd &= ~RxVlan;
-- 
1.7.1